### DISCUSSION TITLE
Encourage Docker installation for non-Pi systems?

### DISCUSSION THREAD

**Discussion by usul0184** [mKeRix/room-assistant#559] [Ideas]
First, this is an awesome project, and I very much appreciate it.

I'm probably an intermediate user of Linux, but I'm very new to Node.js and npm.  So far, I've found them extremely confusing, especially because the current version of Node.js is v15 and room-assistant runs on v12 and v14.  I know there are ways of managing multiple Node.js versions like nvm, but they are way too complicated for the average user.

I'm running room-assistant on general server with lots of other software, and trying to figure out Node.js versioning conflicts is a hopeless task for my level of (in)competence.  I decided to install using Docker, and it was a million times easier.  I realize that Docker is probably not preferable for Raspberry Pi systems because of resource limitations, but Pi systems don't usually have other Node.js software running on them.  Would it make sense to encourage non-Pi users to use Docker in the first instance as the preferred installation method?  In the current documentation, NodeJS is listed first, and it's unclear why anyone should choose one method over another.

**Comment by mKeRix** [comment]
You raise an interesting point! In my mind using NodeJS was easier for users - there is no extra CLI to learn and everything runs on the actual system without limitations (so no file mounting etc. required). Your pros for Docker are all valid too though. I do agree that the docs site is a bit lacking with guidance for people on what installation option to choose.

If you have an idea on how to improve the clarity of the docs on when to choose which option feel free to edit the docs pages and submit a PR. I'm happy to read it over and include the changes into the project. :)

**Reply by BelgarionNL** [reply]
for me docker is so much easier! I also have an unraid server so I am familiar with docker.

Now I am thinking that if I add a bluetooth dongle to my server and put room-assistant on there too I got 1 room already done.
